### ISSUE TITLE
Extract comment blocks and signature

### ISSUE THREAD

**Issue by tdegeus** [doxygen/doxygen#10484]
## Question

I would like to interact with doxygen on a more basic level than e.g. the `xml` output:

**I would like to just have the 'docstrings' (as they are written, i.e. without highlighting) and the function signature as used by doxygen.**

## Motivation
- Debug e.g. broken links. 
- Set the C++ doc-blocks as docstrings for `pybind11` extensions. 

## Example

Consider:
```cpp
namespace name {
/**
 * @brief My first function.
 * @param a This is a parameter.
 */
int foo(int a);

/**
 * This is a class
 */
class Bar
{
public:
    /**
     * @brief Some counter
     * @return integer
     */
    int count();
};
} // namespace name
```

I would like to be able to extract the signature as used by doxygen. For example, I would like to get the following YAML file (though any other format is fine):
```yaml
name::foo(int): |
    @brief My first function.
    @param a This is a parameter.

name::Bar: |
    This is a class

name::Bar::count(): |
    @brief Some counter
    @return integer
```

**Comment by albert-github** [comment]
I don't get what you want

- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by albert-github** [comment]
It would be very difficult to show this information as doxygen does no store the information in this way.

I don't get why an e.g. yaml file with the original information would solve the issues / make it easier to find them?
In the xml file all the relevant information is present and otherwise a warning should have been thrown (or are there cases where a link is expected but no  warning is thrown?)

**Comment by tdegeus** [comment]
Indeed, I have had cases where a link was not created where I did expect it (on implicit creation by putting e.g. `name::foo()` in the above example). It has sometimes been hard to figure out what I had to do to correct. 

My strong motivation is the second listed. My C++ codes have Python bindings written using pybind11. It is very nice to have good docstrings as they are very useful as many editors show them while coding. However, I do not want to write them twice to avoid out-dating them, and I want to keep them in the C++ code not to break my doxygen documentation of it. So what I would like to do is to automatically parse them when I compile the Python API.

Indeed the XML has the info. However, the information that I want is at that point somewhat convoluted. In particular, I am keen to have the doc-block pre formatting.

**Comment by doxygen** [comment]
@tdegeus You could take a look at [this example](https://github.com/doxygen/doxygen/tree/master/addon/doxyapp) that bundles with doxygen. It shows how to build an application using doxygen's parser engine. This does require some understanding of doxygen's internals and may require updates when doxygen's internals change but offers the most flexibility.

**Comment by tdegeus** [comment]
Thanks for the good suggestion @doxygen . As far as I see from reading the code, it contains exactly what I want in terms of symbols -- it seems easy to from it extract a list of symbols and line numbers. Could you also give me a pointer as to how to get the relevant 'docstrings'?

**Comment by doxygen** [comment]
See [Definition::documentation()](https://github.com/doxygen/doxygen/blob/351dba64b602b2334f3d7400af1088c1be62aa99/src/definition.h#L142) and [Definition::briefDescription()](https://github.com/doxygen/doxygen/blob/351dba64b602b2334f3d7400af1088c1be62aa99/src/definition.h#L154)

**Comment by albert-github** [comment]
- Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).
(Issue was closed automatically by GitHub due to the used terminology in the pull request)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
